geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -54,21 +58,24 @@ class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLo
       (intervals.map(_._1.getMillis).min, intervals.map(_._2.getMillis).max)
     }
 
-    // for an attribute query, the primary filters are considered an OR
-    // (an AND would never match unless the attribute is a list...)
-    val propsAndRanges = {
-      val primary = filter.singlePrimary.getOrElse {
-        throw new IllegalStateException("Attribute index does not support Filter.INCLUDE")
-      }
-      getBounds(sft, primary, dates)
-    }
-    val attribute = propsAndRanges.headOption.map(_.attribute).getOrElse {
-      throw new IllegalStateException(s"No ranges found for query filter $filter. " +
-          "This should have been checked during query planning")
+    // TODO currently exclusive ANDs will return nothing - but if it's a list type it could actually match
+    val bounds = AttributeIdxStrategy.getBounds(sft, primary, dates)
+
+    if (bounds.isEmpty) {
+      EmptyPlan(filter)
+    } else {
+      nonEmptyQueryPlan(queryPlanner, hints, bounds)
     }
-    val ranges = propsAndRanges.map(_.range)
+  }
+
+  private def nonEmptyQueryPlan(queryPlanner: QueryPlanner, hints: Hints, bounds: Seq[PropertyBounds]): QueryPlan = {
+    val ds = queryPlanner.ds
+    val sft = queryPlanner.sft
+
+    val attribute = bounds.head.attribute
+    val ranges = bounds.map(_.range)
     // ensure we only have 1 prop we're working on
-    assert(propsAndRanges.forall(_.attribute == attribute))
+    require(bounds.forall(_.attribute == attribute), "Found multiple attributes in attribute filter")
```

Comment:
Can this happen?

Does attr1 = 'val1' and attr2 = 'val2' get passed down there?